### PR TITLE
Omnisci thrift update

### PR DESCRIPTION
**Pull request by xmnlab**
This PR continues the work started at https://github.com/ibis-project/ibis/pull/2256

This PR adds the changes needed to use OmniSciDB>=5.3

cc @jclay

### CODE REVIEW

**Comment by datapythonista** [comment]
Can you use `pymapd=0.24` instead of `>=`. This caused lots of trouble in the CI in the past, I prefer to keep things stable by pinning to the exact version. Thanks!

**Comment by xmnlab** [comment]
@datapythonista sure! I am changing that. thanks

**Comment by jclay** [comment]
@xmnlab thanks for picking this up! Looks good to me.

**Comment by xmnlab** [comment]
@datapythonista, it is ready for a review. thanks!

File: ci/recipe/meta.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ requirements:
     - psycopg2
     - pyarrow >=0.15
     - pydata-google-auth
-    - pymapd =0.23.0  # [py<38]
+    - pymapd 0.24  # [py<38]
```
Comment (jreback) [inline ci/recipe/meta.yaml:46]: so you actually need to fix this version? this caused havoc the last time. I would rather wait until the smoke clears to touch this at all.

**Review by jreback — CHANGES_REQUESTED** [review]

File: ci/recipe/meta.yaml (hunk 1 above)
Comment (xmnlab) [reply ci/recipe/meta.yaml:46]: @datapythonista thanks! yes, that would be great if it would be available for the next ibis release. thanks!

**Comment by jreback** [comment]
@xmnlab can you merge master and let's see the solver times now. also pls add a note in the whatsnew and change setup.py so that the min version of pymapd is enforced.

**Comment by xmnlab** [comment]
PR rebased! the pymapd pinning in setup.py was already there. there was already an entry for the change made here in  release/index.rst ... not sure about the `whatsnew`, where is the file for this? @jreback, thanks for the review.

File: docs/source/backends/omnisci.rst [hunk 2]
```diff
@@ -32,9 +32,7 @@ Create a client by passing in database connection parameters such as ``host``,
        password='HyperInteractive',
    )
 
-Note: OmniSciDB backend support is tested against the latest development
-release of their database using the ``omnisci/core-os-cpu-dev:latest``
-Docker image. Some features may not work on earlier releases.
+Note: OmniSciDB backend support OmniSciDB version greater than or equals to `5.3.0`.
```
Comment (jreback) [inline docs/source/backends/omnisci.rst:35]: make this an actual note, e.g. ```..note::``

File: setup.py [hunk 3]
```diff
@@ -22,7 +22,7 @@
 postgres_requires = sqlite_requires + ['psycopg2']
 mysql_requires = sqlite_requires + ['pymysql']
 
-omniscidb_requires = ['pymapd>=0.12.0']
+omniscidb_requires = ['pymapd==0.24']
```
Comment (jreback) [inline setup.py:25]: >= is probably resasonable here

**Review by jreback — CHANGES_REQUESTED** [review]

File: setup.py (hunk 3 above)
Comment (datapythonista) [reply setup.py:25]: Afaik one equal only works for conda, if I'm not wrong double equal is required here.

File: docs/source/backends/omnisci.rst (hunk 2 above)
Comment (xmnlab) [reply docs/source/backends/omnisci.rst]: done. thanks!

**Comment by xmnlab** [comment]
it is ready for a new round of review. thanks!

**Comment by jreback** [comment]
@datapythonista any idea here why conda solves take 75 min.

**Comment by datapythonista** [comment]
> @datapythonista any idea here why conda solves take 75 min.

Because the conda solver has a `time.sleep(random.random())` on it? ;)

No idea, seems still less than the last PRs, but I think the previous one was 40 minutes, it's quite a huge difference. We can migrate to mamba or something, but I think we're very close to move the omnisci backend outside, so unless anyone else want to check it, I think we can survive with this nonsense for a bit more. But in GitHub actions we're pinning as much as we can, so mamba seems the only option.

**Comment by jreback** [comment]
> > @datapythonista any idea here why conda solves take 75 min.
> 
> Because the conda solver has a `time.sleep(random.random())` on it? ;)
> 
> No idea, seems still less than the last PRs, but I think the previous one was 40 minutes, it's quite a huge difference. We can migrate to mamba or something, but I think we're very close to move the omnisci backend outside, so unless anyone else want to check it, I think we can survive with this nonsense for a bit more. But in GitHub actions we're pinning as much as we can, so mamba seems the only option.

can we just pause this build for rn? (or remove pymapd)? 

actually a more general issue is why do we install all of the optionals in the conda build? we should not do this anywhere (even and espcially the doc build), there is NO reason to. just the basic stuff.

**Comment by jreback** [comment]
> Because the conda solver has a `time.sleep(random.random())` on it? ;)

this is not the reason. this recipe should solve in seconds. The problem is we have too much incompatible stuff in it.

**Comment by jreback** [comment]
let's see this run again

**Comment by xmnlab** [comment]
```
Specifications:

  - conda=4.6 -> python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0']

Your python: python=3.8

```

it seems the python version used by conda on the windows environment is 3.8 and it is conflicting with the conda version pinned (4.6).

I will push it to 4.7, it seems it has support for python 3.8

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @xmnlab 
ok we *really* need to fix the conda solve time.

**Comment by xmnlab** [comment]
thanks @jreback, yes that sounds reasonable.

**Pull request merged**